lib/geoengineer/project.rb

```diff
@@ -12,16 +12,25 @@ class GeoEngineer::Project
   include HasValidations
 
   attr_accessor :org, :name
+
+  ## Environment is an object of type GeoEngineer::Environment
   attr_reader :environment
 
-  validate -> { environments.nil? ? "Project #{full_name} must have an environment" : nil }
+  ## Environments is available from @_attributes generated by HasAttributes as
+  # an Array of environment names ["test","development"...]
+  validate -> { environments.nil? ? "Project #{full_name} must have an environments attribute" : nil }
   validate -> { all_resources.map(&:errors).flatten }
 
   def initialize(org, name, environment, &block)
     @org = org
     @name = name
     @environment = environment
     instance_exec(self, &block) if block_given?
+    ## environments is an attribute that needs to be set, so lets make sure to set it.
+    if !@environment.nil? &&
+       !@environment.name
+      @_attributes[environments] = [@environment.name]
```

Comment:
Reading this looks a little odd—you're accessing a hash (`@_attributes`) using what is potentially an array (`environments`) as its key. Is that correct?

Reply:
That is correct. From inspecting the object in generated in tests "@_attributes={\"environments\"=>\"test\"}>}" we get this.

Reply:
hmm, may be not, in the tests it is a string.

Reply:
I might be misunderstanding you, but `@_attributes={"environments"=>"test"}>` means that you should write:
```ruby
@_attributes['environments']
```
...not:
```ruby
@_attributes[environments]
```
Does that clarify what I'm getting at?

Reply:
yes

Reply:
it was a typo